src/foam/nanos/crunch/lite/MinMaxCapabilityRefinement.js

```diff
@@ -51,15 +52,31 @@ foam.CLASS({
             case APPROVED:
               numberPending++;
               continue;
+            case REJECTED:
+              numberRejected++;
+              continue;
           }
         }
 
+        int numberTotal = numberGranted + numberPending + numberRejected;
+
+        int numberGrantedOrPending = numberGranted + numberPending;
+
+        // TODO: right now this represents the capablePayloads haven't been processed for preqreqs yet
+        if ( numberTotal == 0 ){
+          return CapabilityJunctionStatus.ACTION_REQUIRED;
+        }
+
+        if ( getMin() > numberGrantedOrPending ){
```

Comment:
Don't understand this one.

Reply:
NumberGrantedorPending are basically the available Payloads which are GRANTED or can eventually be turned into GRANTED. If the MIN is greater than the number of available payloads which are GRANTED or can eventually be turned into GRANTED, then it is impossible for the total amount of GRANTED payloads to be greater than the MIN, thereby fulfilling the minimum requirement.

Imagine theres a min max which has 10 prerequisites attached and a min of 2.

If the user selected only 3 of those prereqs in the wizard, then he will get 3 approvals generated all in PENDING state.

If 1 of those 3 get rejected. Then we will have 2 numberGrantedOrPending which could still potentially satisfy the min requirement of 2  if those 2 get approved and thereby set to GRANTED.

If 2 of those 3 get rejected. then we will have 1 numberGrantedOrPending which would be impossible to satisfy the min requirement of 2 even if that 1 payload is GRANTED.